Framework/Algorithms/src/InterpolatingRebin.cpp

```diff
@@ -200,27 +201,18 @@ void InterpolatingRebin::outputYandEValues(
 *  @throw invalid_argument if any output x-values are outside the range of input
 *x-values
 **/
-void InterpolatingRebin::cubicInterpolation(const HistogramData::BinEdges &xOld,
-                                            const MantidVec &yOld,
-                                            const MantidVec &eOld,
-                                            const HistogramData::BinEdges &xNew,
-                                            MantidVec &yNew,
-                                            MantidVec &eNew) const {
-  // Make sure y and e vectors are of correct sizes
+void InterpolatingRebin::cubicInterpolation(const Histogram &oldHistogram,
```

Comment:
@DimitarTasev  For example:

```cpp
Histogram InterpolatingRebin::cubicInterpolation(const Histogram &oldHistogram, const BinEdges &xNew) const {
// ...
   if (!canInterpol) {
     if (VectorHelper::isConstantValue(yOld.rawData())) {
       // this copies the single y-value into the output array, errors are still
       // calculated from the nearest input data points
       // this is as much as we need to do in this (trival) case
      return noInterpolation(oldHistogram, xNew);
     } else { // some points are two close to the edge of the data
// ...
}
```

The size of `outputW->mutableY` and `outputW->mutableE` is simply the size of `xNew` minus 1.